common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -60,61 +61,15 @@ public int compare(CharSequence o1, CharSequence o2) {
             if (o1 == o2) {
                 return 0;
             }
-
-            AsciiString a1 = o1 instanceof AsciiString ? (AsciiString) o1 : null;
-            AsciiString a2 = o2 instanceof AsciiString ? (AsciiString) o2 : null;
-
-            int result;
             int length1 = o1.length();
             int length2 = o2.length();
             int minLength = Math.min(length1, length2);
-            if (a1 != null && a2 != null) {
-                byte[] thisValue = a1.value;
-                byte[] thatValue = a2.value;
-                for (int i = 0; i < minLength; i++) {
-                    byte v1 = thisValue[i];
-                    byte v2 = thatValue[i];
-                    if (v1 == v2) {
-                        continue;
-                    }
-                    int c1 = toLowerCase(v1);
-                    int c2 = toLowerCase(v2);
-                    result = c1 - c2;
-                    if (result != 0) {
-                        return result;
-                    }
-                }
-            } else if (a1 != null) {
-                byte[] thisValue = a1.value;
-                for (int i = 0; i < minLength; i++) {
-                    int c1 = toLowerCase(thisValue[i]);
-                    int c2 = toLowerCase(o2.charAt(i));
-                    result = c1 - c2;
-                    if (result != 0) {
-                        return result;
-                    }
-                }
-            } else if (a2 != null) {
-                byte[] thatValue = a2.value;
-                for (int i = 0; i < minLength; i++) {
-                    int c1 = toLowerCase(o1.charAt(i));
-                    int c2 = toLowerCase(thatValue[i]);
-                    result = c1 - c2;
-                    if (result != 0) {
-                        return result;
-                    }
-                }
-            } else {
-                for (int i = 0; i < minLength; i++) {
-                    int c1 = toLowerCase(o1.charAt(i));
-                    int c2 = toLowerCase(o2.charAt(i));
-                    result = c1 - c2;
-                    if (result != 0) {
-                        return result;
-                    }
+            for (int i = 0; i < minLength; i++) {
+                int result = toLowerCase(o1.charAt(i)) - toLowerCase(o2.charAt(i));
```

Comment:
To clarify...we can pull this in now, but I have a PR upcoming that will likely just restore it.  So why make the merge more difficult?

Sorry if I'm slow in this one...feel like I'm missing something.  Can we not leave this as it is because it is broken or this is a bug?